utils/windows-utils.ts

```diff
@@ -11,4 +14,77 @@ export function validatePackageID(id: string) {
   else {
     throw (`Package ID should be atleast 3 characters long: ${id}`)
   }
+}
+
+const DISALLOWED_WINDOWS_PACKAGE_CHARS_REGEX = /[^a-zA-Z0-9_]/g;
+
+export function generateWindowsPackageId(host: string): string {
+  const parts = host
+    .split(".")
+    .reverse()
+    .map(p => p.trim().toLowerCase())
+    .filter(p => p.length > 0)
+    .map(p => p.replace(DISALLOWED_WINDOWS_PACKAGE_CHARS_REGEX, "-"))
+  parts.push("edge");
+  return parts.join("-");
+}
+
+export function validateWindowsOptions(options) {
+  const validationErrors: { field: keyof WindowsPackageOptions | null, error: string }[] = [];
+  if (!options) {
+    validationErrors.push({ field: null, error: "No options specified " });
+    return validationErrors;
+  }
+
+  if (!options.packageId) {
+    validationErrors.push({ field: "packageId", error: "No package ID" });
+  }
+
+  if (!options.name || options.name.trim().length === 0) {
+    validationErrors.push({ field: "name", error: "Must have a valid app name" });
+  }
+
+  if (!options.version || options.version.trim().length === 0 || options.version.trim().length < 2) {
+    validationErrors.push({ field: "version", error: "Must have a valid app version" });
+  }
+
+  // Validating signing options when we have a signing mode 
+  if (options.publisher) {
+    if (!options.publisher) {
```

Comment:
I think this is a bug - the first if statement checks if publisher is truthy, then inner if statement checks if it's not truthy.